ewfs/noise_models.py

```diff
@@ -45,3 +55,49 @@ def bitflip_model(p: float) -> NoiseModel:
 
     return noise_bit_flip
 
+
+def get_single_qubit_gates(circ: QuantumCircuit) -> Set[str]:
```

Comment:
[nit]: We could use native types instead of using `Set` from the `typing` module:

```suggestion
def get_single_qubit_gates(circ: QuantumCircuit) -> set[str]:
```